minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Utils/CraftingHelper.java

```diff
@@ -882,4 +880,58 @@ public static void dumpMinecraftObjectRules(String filename) {
             e.printStackTrace();
         }
     }
+
+    /**
+     * Utility method to auto-generate item, block, and recipe lists as individual json arrays
+     *
+     * @param filename location to save the dumped json file.
+     */
+    public static void dumpVoxelObs(String filename) {
+        JsonObject obs = new JsonObject();
+        JSONWorldDataHelper.GridDimensions gridDimensions = new JSONWorldDataHelper.GridDimensions(3, 3, 3);
+        JSONWorldDataHelper.buildGridData(obs, gridDimensions, Objects.requireNonNull(Minecraft.getMinecraft().getIntegratedServer()).getPlayerList().getPlayerByUUID(Minecraft.getMinecraft().player.getUniqueID()), "voxels");
+        try {
+            Writer writer = new FileWriter(filename);
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            gson.toJson(obs, writer);
+            System.out.println("Wrote json to " + System.getProperty("user.dir") + filename);
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static RayOffset makeOffset(float pitch, float yaw, float magnitude){
+        RayOffset offset = new RayOffset();
+        offset.setPitch(pitch);
+        offset.setYaw(yaw);
+        offset.setDistance(magnitude);
+        return offset;
+    }
+
+    /**
+     * Utility method to auto-generate item, block, and recipe lists as individual json arrays
+     *
+     * @param filename location to save the dumped json file.
+     */
+    public static void dumpRichLidarObs(String filename) {
+        JsonObject obs = new JsonObject();
+        List<RayOffset> rayOffsets = new ArrayList<RayOffset>(){{
```

Comment:
this is for debugging only yes?

Reply:
debugging only - would like to leave for now while I am still hacking